applications/file-split-ftp/src/main/java/org/springframework/integration/samples/filesplit/Application.java

```diff
@@ -0,0 +1,238 @@
+
+	@Bean
+	public IntegrationFlow fromFile() {
+		return IntegrationFlows.from(Files.inboundAdapter(new File("/tmp/in"))
+					.patternFilter("*.txt"), e -> e.poller(Pollers.fixedDelay(5000)))
+				.enrichHeaders(h -> h.header(MessageHeaders.ERROR_CHANNEL, "tfrErrors.input"))
+				.handle(Files.splitter(true, true))
+				.route(ptr())
+				.get();
+	}
+
+	private PayloadTypeRouter ptr() {
+		PayloadTypeRouter payloadTypeRouter = new PayloadTypeRouter();
+		payloadTypeRouter.setChannelMappings(new StringStringMapBuilder()
+				.put(FileSplitter.FileMarker.class.getName(), "markers.input")
+				.put(String.class.getName(), "lines.input")
+				.get());
+		return payloadTypeRouter;
+	}
+
+	/**
+	 * Process lines; append (no flush) to the appropriate file.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow lines() {
+		return f -> f.enrichHeaders(h -> h.headerExpression(
+						FileHeaders.FILENAME, "payload.substring(1, 4) + '.txt'", true))
+				.handle(Files.outboundAdapter("'/tmp/out'")
+						.appendNewLine(true)
+						.fileExistsMode(FileExistsMode.APPEND_NO_FLUSH), // files remain open for efficiency
+							e -> e.id("fileOut"));
+	}
+
+	/**
+	 * Process file markers; ignore START, when END, flush the files, ftp them and
+	 * send an email.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow markers() {
+		return f -> f.<FileSplitter.FileMarker>filter(m -> m.getMark().equals(FileSplitter.FileMarker.Mark.END),
+						e -> e.id("markerFilter").discardChannel("nullChannel")) // INT-4109
+				.publishSubscribeChannel(s -> s
+
+						// first trigger file flushes
+						.subscribe(sf -> sf.transform("'/tmp/out/.*\\.txt'")
+										.handle("fileOut.handler", "trigger", e -> e.id("flusher")))
+
+						// send the first file
+						.subscribe(sf -> sf.<FileSplitter.FileMarker, File>transform(p -> new File("/tmp/out/002.txt"))
+								.enrichHeaders(h -> h.header(FileHeaders.FILENAME, "002.txt", true))
+								.handleWithAdapter(a -> a.ftp(ftp1()).remoteDirectory("foo"), e -> e.id("ftp002")))
+
+						// send the second file
+						.subscribe(sf -> sf.<FileSplitter.FileMarker, File>transform(p -> new File("/tmp/out/006.txt"))
+								.enrichHeaders(h -> h.header(FileHeaders.FILENAME, "006.txt", true))
+								.handleWithAdapter(a -> a.ftp(ftp2()).remoteDirectory("foo"), e -> e.id("ftp006")))
+
+						// send the third file
+						.subscribe(sf -> sf.<FileSplitter.FileMarker, File>transform(p -> new File("/tmp/out/009.txt"))
+								.enrichHeaders(h -> h.header(FileHeaders.FILENAME, "009.txt", true))
+								.handleWithAdapter(a -> a.ftp(ftp3()).remoteDirectory("foo"), e -> e.id("ftp009")))
+
+						// send an email
+						.subscribe(sf -> sf.<FileSplitter.FileMarker, String>transform(p -> p.getFilePath())
+								.enrichHeaders(Mail.headers()
+										.subject("File successfully split and transferred")
+										.from("foo@bar")
+										.toFunction(m -> new String[] {"bar@baz"}))
```

Comment:
Yep! These are up to `@ConfigurationProperties`. Our own custom one.
I mean for the real mail server tests.
But I see we have so many of options to expose...

Maybe we can live without that for now...